emmet-core/emmet/core/qc_tasks.py

```diff
@@ -0,0 +1,609 @@
+
+        Parameters
+        ----------
+        calc_doc
+            A QChem calculation document.
+
+        Returns
+        --------
+        InputDoc
+            A summary of the input molecule and corresponding calculation parameters
+        """
+        # TODO : modify this to get the different variables from the task doc.
+        return cls(
+            initial_molecule=calc_doc.input.initial_molecule,
+            rem=calc_doc.input.rem,
+            # lev_theory = level_of_theory(calc_doc.input.rem),
+            task_type=calc_doc.task_type,
+            tags=calc_doc.input.tags,
+            solvent=calc_doc.input.solvent,
+            # lot_solvent_string = calc_doc.input.lot_solvent_string,
+            # custom_smd = calc_doc.input.custom_smd,
+            # special_run_type = calc_doc.input.special_run_type,
+            # smiles = calc_doc.input.smiles,
+            calc_spec=calc_doc.calc_type,
+        )
+
+
+class CustodianDoc(BaseModel):
+    corrections: List[Any] = Field(
+        None,
+        title="Custodian Corrections",
+        description="List of custodian correction data for calculation.",
+    )
+
+    job: dict = Field(
+        None,
+        title="Custodian Job Data",
+        description="Job data logged by custodian.",
+    )
+
+
+# AnalysisDoc? Is there a scope for AnalysisDoc in QChem?
+
+
+class TaskDoc(MoleculeMetadata):
+    """
+    Calculation-level details about QChem calculations that would eventually take over the TaskDocument implementation
+    """
+
+    dir_name: str = Field(None, description="The directory for this QChem task")
+
+    state: QChemStatus = Field(None, description="State of this QChem calculation")
+
+    calcs_reversed: List[Calculation] = Field(
+        None,
+        title="Calcs reversed data",
+        description="Detailed data for each QChem calculation contributing to the task document.",
+    )
+
+    molecule: Molecule = Field(
+        None, description="Final optimized/output molecule from this task"
+    )
+
+    task_type: Union[CalcType, TaskType] = Field(
+        None, description="the type of QChem calculation"
+    )
+
+    orig: Dict[str, Any] = Field(
+        {}, description="Summary of the original Q-Chem inputs"
+    )
+
+    input: InputDoc = Field(
+        None,
+        description="The input molecule and calc parameters used to generate the current task document.",
+    )
+
+    output: OutputDoc = Field(
+        None,
+        description="The exact set of output parameters used to generate the current task document.",
+    )
+
+    # TODO: Implement entry dict
+
+    custodian: List[CustodianDoc] = Field(
+        None,
+        title="Calcs reversed data",
+        description="Detailed custodian data for each QChem calculation contributing to the task document.",
+    )
+
+    critic2: Dict[str, Any] = Field(
+        None, description="Outputs from the critic2 calculation if performed"
+    )
```

Comment:
Going forward, there's really no reason that this needs to be top-level. It could just be within output. But it seems like you're trying to stick with the current (e.g. atomate1) format, in which case of course this should stay.